### PR TITLE
Upgrade merkle proofs API

### PR DESCRIPTION
**Pull request by pavelkalinnikov**
This change switches to the new proof generation API in `merkle` repo.

### CODE REVIEW

File: server/proof_fetcher_test.go [hunk 1]
```diff
@@ -99,12 +99,12 @@ func TestTree32InclusionProofFetchAll(t *testing.T) {
...
 
 				if got, want := len(proof.Hashes), len(refProof); got != want {
-					t.Fatalf("(%d, %d, %d): got proof len: %d, want: %d: %v\n%v", ts, s, l, got, want, fetches, refProof)
+					t.Fatalf("(%d, %d, %d): got proof len: %d, want: %d: %v\n%v", ts, s, l, got, want, nodes, refProof)
 				}
 
 				for i := 0; i < len(proof.Hashes); i++ {
 					if got, want := hex.EncodeToString(proof.Hashes[i]), hex.EncodeToString(refProof[i].Value.Hash()); got != want {
-						t.Fatalf("(%d, %d, %d): %d got proof node: %s, want: %s l:%d fetches: %v", ts, s, l, i, got, want, len(proof.Hashes), fetches)
+						t.Fatalf("(%d, %d, %d): %d got proof node: %s, want: %s l:%d nodes: %v", s, ts, l, i, got, want, len(proof.Hashes), nodes)
```
Comment (AlCutter) [inline server/proof_fetcher_test.go:107]: What was the reasoning to swap `ts` and `s` around here (I think the original ordering was based on the nested loop vars: outer loop is `ts`, then `s`, then `l`) ?

**Review by AlCutter — APPROVED** [review]
🎯  just one little question inline

File: server/proof_fetcher_test.go (hunk 1 above)
Comment (pavelkalinnikov) [reply server/proof_fetcher_test.go]: Restored the original order.

File: go.sum [hunk 2]
```diff
@@ -1121,6 +1121,7 @@ golang.org/x/sys v0.0.0-20210917161153-d61c044b1678/go.mod h1:oPkhp1MJrh7nUepCBc
 golang.org/x/sys v0.0.0-20211007075335-d3039528d8ac/go.mod h1:oPkhp1MJrh7nUepCBck5+mAzfO9JrbApNNgaTdGDITg=
 golang.org/x/sys v0.0.0-20211019181941-9d821ace8654/go.mod h1:oPkhp1MJrh7nUepCBck5+mAzfO9JrbApNNgaTdGDITg=
 golang.org/x/sys v0.0.0-20211124211545-fe61309f8881/go.mod h1:oPkhp1MJrh7nUepCBck5+mAzfO9JrbApNNgaTdGDITg=
+golang.org/x/sys v0.0.0-20211210111614-af8b64212486/go.mod h1:oPkhp1MJrh7nUepCBck5+mAzfO9JrbApNNgaTdGDITg=
```
Comment (pphaneuf) [inline go.sum:1124]: Is that some unrelated stuff? Should there be some `go mod tidy` done?

**Review by pphaneuf — APPROVED** [review]
